app/src/main/java/com/example/photofy/activities/LoginActivity.java

```diff
@@ -1,5 +1,7 @@
 package com.example.photofy.activities;
 
+import static com.example.photofy.activities.SpotifyLoginActivity.spotifyToken;
```

Comment:
this will not work, lifecycle of activities or any other UI component is managed by Android and may destroy an activity once it is removed from the stack/not visible. Consider using Android ViewModel to store data whose lifecycle you wants to manage.

Reply:
Would passing the token as an extra through an intent work as well? Also, would variables defined in the PhotofyApplication (extends Application) class be always visible? I extracted private environmental variables (API keys) through the PhotofyApplication class. It is not an activity, so I don't believe the activity lifecycle/onDestroy would apply.

Reply:
Yes, both options will work. Passing the token or any data between two activities or keeping the reference in the PhotofyApplication.

Passing through Intent will become very cumbersome very quickly. For example, spotify auth token will be needed in multiple activities, You get that token in `SpotifyLoginActivity`. You will need to pass it to `MainActivity` -> 'CameraFragment' -> 'CameraActionFragment' -> SongsPreview -> PostActivity; Similarly you will need the token in FeedsActvity etc.

Keeping the model hierarchy in ApplicationContext is a fair compromise. You will be managing the lifecycle of the data by yourself along with  splitting the logic to fetch the data in some other place and set/get in ApplicationContext. That's fine for your usecase. We can come back to evaluating ViewModel or any other code quality measure later once we have better handle on the functionality.